src/core/url-beautifier/search/generator.ts

```diff
@@ -65,80 +109,71 @@ export default class SearchUrlGenerator extends UrlGenerator<UrlBeautifier.Searc
     return path;
   }
 
-  buildUrl(path: string[], query: object) {
-    let url = `/${path.map((part) => encodeURIComponent(part)).join('/')}`;
-
-    if (this.config.suffix) {
-      url += `/${this.config.suffix.replace(/^\/+/, '')}`;
-    }
-
-    const queryPart = Object.keys(query)
-      .sort()
-      .map((key) => `${key}=${encodeURIComponent(query[key])}`)
-      .join('&');
-
-    if (queryPart) {
-      url += '?' + queryPart;
-    }
-
-    return url.replace(/\s|%20/g, '-');
+  static convertSort(sort: Store.Sort) {
+    const { field, descending = !!sort.descending } = sort;
+    return [field, [descending]];
   }
 
-  static convertPathRefinements({ refinements }: UrlBeautifier.SearchRequest) {
+  static convertPathRefinements(refinements: UrlBeautifier.Refinement[]) {
     const path = [];
     const valueRefinements = [];
 
     for (let i = refinements.length - 1; i >= 0; --i) {
-      if (refinements[i].type === 'Value') {
+      if (refinements[i]['value']) {
         valueRefinements.push(...refinements.splice(i, 1));
       }
     }
 
-    valueRefinements.map(SearchUrlGenerator.convertToSelectedValueRefinement)
-      .sort(SearchUrlGenerator.refinementsComparator)
+    valueRefinements.sort(SearchUrlGenerator.refinementsComparator)
       .forEach((selectedValueRefinement) =>
-        path.push(selectedValueRefinement.value, selectedValueRefinement.navigationName));
+        path.push(utils.encodeChars(selectedValueRefinement.value), utils.encodeChars(selectedValueRefinement.field)));
```

Comment:
would we be able to forgo encoding these individually and instead map the returned `path` like ?
```
return path.map(utils.encodeChars);
```